src/app/activity/bounty/Paid.ts

```diff
@@ -0,0 +1,180 @@
+		payerMessage = await bountyChannel.messages.fetch(messageId).catch(e => {
+			LogUtils.logError(`could not find IOU ${request.bountyId} in channel ${channelId} in guild ${request.guildId}`, e);
+			throw new RuntimeError(e);
+		});
+    } else {
+        payerMessage = request.message;
+    }
+
+	const bountyUrl = process.env.BOUNTY_BOARD_URL + request.bountyId;
+	const owedToUser: GuildMember = await paidByUser.guild.members.fetch(getDbResult.dbBountyResult.owedTo.discordId);
+
+    
+    await paidBountyMessage(getDbResult.dbBountyResult, payerMessage, paidByUser, owedToUser);
+	
+	const creatorPaidDM = 
+        `Thank you for marking your IOU as paid ${bountyUrl}\n` +
+        `If you haven't already, please remember to tip <@${owedToUser.id}>`;
+
+    
+    await paidByUser.send({ content: creatorPaidDM });
+    return;
+}
+
+/**
+ * Wraps read only calls to the database.
+ * Intended to be replaced with calls to the API.
+ * @param request PaidRequest, passed from activity initiator
+ * @returns 
+ */
+const getDbHandler = async (request: PaidRequest): Promise<{dbBountyResult: BountyCollection, bountyChannel: string}> => {
+    const db: Db = await MongoDbUtils.connect('bountyboard');
+	const bountyCollection = db.collection('bounties');
+    const customerCollection = db.collection('customers');
+
+	const dbBountyResult: BountyCollection = await bountyCollection.findOne({
+		_id: new mongo.ObjectId(request.bountyId),
+		status: BountyStatus.open,
+	});
+
+    if (request.message) {
+        return {
+            dbBountyResult: dbBountyResult,
+            bountyChannel: null
+        }
+    }
+
+    const dbCustomerResult: CustomerCollection = await customerCollection.findOne({
+        customerId: request.guildId,
+    });
+
+    return {
+        dbBountyResult: dbBountyResult,
+        bountyChannel: dbCustomerResult.bountyChannel
+    }
+}
+
+// TODO: consider adding the previous read result as a parameter to save a db read
+const writeDbHandler = async (request: PaidRequest, paidByUser: GuildMember): Promise<void> => {
+    const db: Db = await MongoDbUtils.connect('bountyboard');
+	const bountyCollection = db.collection('bounties');
+
+	const dbBountyResult: BountyCollection = await bountyCollection.findOne({
+		_id: new mongo.ObjectId(request.bountyId),
+		status: BountyStatus.open,
+	});
+
+	const currentDate = (new Date()).toISOString();
+	const writeResult: UpdateWriteOpResult = await bountyCollection.updateOne(dbBountyResult, {
+		$set: {
+			reviewedBy: {
+				discordHandle: paidByUser.user.tag,
+				discordId: paidByUser.user.id,
+				iconUrl: paidByUser.user.avatarURL(),
+			},
+            // TO-DO: What is the point of status history if we publish createdAt, claimedAt... as first class fields?
+            // note that createdAt, claimedAt are not part of the BountyCollection type
+			reviewedAt: currentDate,
+			status: BountyStatus.complete,
```

Comment:
I think adding a `paid` flag as a boolean will help us implement paid status for non-IOU bounties in the future

Reply:
Same question as above - a new status? Right now it's inferring Completed + IOU => Paid.

Reply:
I would separate them even further logically:  
`status` and `paidStatus` both as top level fields for the bounty data schema. 

`paidStatus` can get its values from a constant like `BountyPaidStatus` with string fields `paid` and `unpaid`

Reply:
Done